cmd/dockerd/daemon.go

```diff
@@ -611,8 +611,8 @@ func loadDaemonCliConfig(opts *daemonOptions) (*config.Config, error) {
 		// If CDISpecDirs is set to an empty string, we clear it to ensure that CDI is disabled.
 		conf.CDISpecDirs = nil
 	}
-	if !conf.Experimental {
-		// If experimental mode is not set, we clear the CDISpecDirs to ensure that CDI is disabled.
+	if !conf.Features["cdi"] {
```

Comment:
Adding some context to this PR, and looking for feedback.

The motivation for this PR was that a team at Docker was working on a feature for which they are considering CDI; while discussing that approach, we found that

- `experimental` was a bit of a big hammer and, as we already have the `features` option in the daemon config, having more granular options seemed like a better choice
- more so; if `cdi` is enabled by default in the future, then having the `cdi` option in "features" will just become just a "no-op"
- whereas keeping `experimental` enabled in the config may (with updates of the daemon) result in "other, new, experimental features to be enabled", which may not always be desirable.

There's one part I could use input on; while we have not shipped this feature in a stable release yet, there _MAY_ be users who already started trying CDI, and have `experimental` enabled for that reason. Technically, we don't have to be concerned about that (hey, it's experimental), but if that's a concern, we could consider making this "either" experimental is enabled (meaning: enable "all the experimental things"), or "cdi enabled in features";

```suggestion
	if !conf.Features["cdi"] &&  !conf.Experimental {
		// If the CDI feature is not enabled, we clear the CDISpecDirs to ensure that CDI is disabled.
```

Alternatively, if we want to make that explicit, we could choose to make `conf.Experimental` automatically append `cdi: true` to the features list (if it's not set by the user, i.e.; `cdi: false` should still be taken into account).

Reply:
I'd be in favor of not complicating things too much. The feature is quite new so I think we can still afford to make a breaking change there. It also should be easily visible to any concerned user.

We should however note that explicitly in the release notes/documentation (@dvdksn).

Reply:
Yup, it's not released yet, so "anything goes", and I added a `docs/revisit` label because we need to check if we already documented this with "experimental" (at least we need to update the `dockerd` reference to include/mention the feature option.

After this is merged, I'll also leave a quick comment on the original PRs (in case someone visits those pages in future, and now being confused it doesn't work after setting `experimental`)